src/js/decafmud.js

```diff
@@ -1598,17 +1641,25 @@ DecafMUD.prototype.encode = function(data) {
 /** Read through data, only stopping for TELNET sequences. Pass data through
  *  towards the display handler. */
 DecafMUD.prototype.processBuffer = function() {
-	var enc, data, ind, out;
-
-	data = this.inbuf.join(''), IAC = DecafMUD.TN.IAC, left='';
+	var enc, data, ind, out, compressBuf, temp;
+	data = [];
+	// Looks nicer than this.inbuf.map(s=>Array.from(s)).flat().map(s=>String.fromCharCode(s)).join('');
```

Comment:
No need to leave a comment with why this looks nicer. I would instead explain why it's sometimes an array or a string

Reply:
https://github.com/Tiger-Tom/DecafMUD/commit/aa44499e6c0fc0d9a8b96a250f20dc26930bee3d